File: internal/service/base/accounts.go
package base

import (
	"context"

	pb "ai-rag-demo/api/base/v1"
	"ai-rag-demo/internal/biz/base"
)

type AccountService struct {
	pb.UnimplementedAccountsServer
	accountBiz *base.AccountBiz
}

func NewAccountService(
	accountBiz *base.AccountBiz,
) *AccountService {
	return &AccountService{
		accountBiz: accountBiz,
	}
}

func (s *AccountService) ChangePassword(ctx context.Context, req *pb.ChangePasswordRequest) (*pb.ChangePasswordResponse, error) {
	return s.accountBiz.ChangePassword(ctx, req)
}
func (s *AccountService) GetUserInfo(ctx context.Context, req *pb.GetUserInfoRequest) (*pb.Account, error) {
	return s.accountBiz.GetUserInfo(ctx, req)
}
func (s *AccountService) Login(ctx context.Context, req *pb.LoginRequest) (*pb.LoginResponse, error) {
	return s.accountBiz.Login(ctx, req)
}
func (s *AccountService) Logout(ctx context.Context, req *pb.LogoutRequest) (*pb.LogoutResponse, error) {
	return s.accountBiz.Logout(ctx, req)
}
func (s *AccountService) Register(ctx context.Context, req *pb.RegisterRequest) (*pb.RegisterResponse, error) {
	return s.accountBiz.Register(ctx, req)
}

File: web/src/api/accounts.ts
import { http } from './http';
import type {
  RegisterRequest,
  RegisterResponse,
  LoginRequest,
  LoginResponse,
  ChangePasswordRequest,
  Account,
} from '../types/api';

export const accountsApi = {
  // 注册账号
  register(data: RegisterRequest): Promise<RegisterResponse> {
    return http.post('/base/v1/accounts/register', data);
  },

  // 登录
  login(data: LoginRequest): Promise<LoginResponse> {
    return http.post('/base/v1/accounts/login', data);
  },

  // 登出
  logout(): Promise<void> {
    return http.post('/base/v1/accounts/logout');
  },

  // 修改密码
  changePassword(data: ChangePasswordRequest): Promise<void> {
    return http.post('/base/v1/accounts/changePassword', data);
  },

  // 获取用户信息
  getUserInfo(): Promise<Account> {
    return http.get('/base/v1/accounts/userInfo');
  },
};
